### PR TITLE
Remove check for J2SE_18 assuming Java 8 is the lowest level supported

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Remove check for `J2SE_18` assuming `Java 8` is the lowest level supported

This assumes that `OpenJ9` builds `Java 8` and upward Java levels.
Note: some `J2SE_1[5|6|7]` references are still around to provide back compatible support.

Reviewer @pshipton
FYI: @danheidinga

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -2547,9 +2547,7 @@ modifyDllLoadTable(J9JavaVM * vm, J9Pool* loadTable, J9VMInitArgs* j9vm_args)
 			jvmti = TRUE;
 		} else
 #if defined(J9VM_OPT_SIDECAR)
-		if ((J2SE_VERSION(vm) & J2SE_VERSION_MASK) >= J2SE_16) {
-			jvmti = TRUE;
-		}
+		jvmti = TRUE;
```
Comment (JasonFengJ9) [inline runtime/vm/jvminit.c]: Note: this code can be further simplified if `J9VM_OPT_SIDECAR`is always enabled which is the case for all current specs but leave that to another PR instead.

File: runtime/vm/BytecodeInterpreter.hpp [hunk 2]
```diff
@@ -4281,11 +4273,7 @@ done:;
 		walkState->userData1 = (void*)(UDATA)(depth + 1); /* Skip the local INL frame */
 		walkState->userData2 = NULL;
 		walkState->userData3 = (void *) FALSE;
-		walkState->frameWalkFunction = cInterpGetStackClassIterator;
-
-		if ((J2SE_VERSION(_vm) & J2SE_VERSION_MASK) >= J2SE_18) {
-			walkState->frameWalkFunction = cInterpGetStackClassJEP176Iterator;
-		}
+		walkState->frameWalkFunction = cInterpGetStackClassJEP176Iterator;
```
Comment (pshipton) [inline runtime/vm/BytecodeInterpreter.hpp]: Seems the function cInterpGetStackClassIterator can be deleted now.

File: runtime/vm/jvminit.c (hunk 1 above)
Comment (pshipton) [reply runtime/vm/jvminit.c]: Is there some reason not to just delete all this code and replace it with `jvmti = TRUE;`? Seems an easy change.

Comment (JasonFengJ9) [reply runtime/vm/jvminit.c]: As per previous note, I preferred to leave further simplification to another PR for the flag `J9VM_OPT_SIDECAR`.
However just noticed that original code already relies on `J9VM_OPT_SIDECAR` is on, other syntax error occurs. 

```
#if defined(J9VM_OPT_JVMTI)
		if ((strstr(testString, VMOPT_AGENTLIB_COLON)==testString) || (strstr(testString, VMOPT_AGENTPATH_COLON)==testString)) {
			ADD_FLAG_AT_INDEX( ARG_REQUIRES_LIBRARY, i );
			jvmti = TRUE;
		} else   ---> following code won't exist if J9VM_OPT_SIDECAR is not enabled.
#if defined(J9VM_OPT_SIDECAR)
		if ((J2SE_VERSION(vm) & J2SE_VERSION_MASK) >= J2SE_16) {
			jvmti = TRUE;
		}
#endif
#endif
```

Will remove `jvmti` completely instead.

File: runtime/vm/BytecodeInterpreter.hpp (hunk 2 above)
Comment (JasonFengJ9) [reply runtime/vm/BytecodeInterpreter.hpp]: Deleted. Pls have another look.

**Comment by pshipton** [comment]
Copyright check failed

**Comment by JasonFengJ9** [comment]
Copyright has been fixed. It passes now.

**Comment by pshipton** [comment]
jenkins test sanity plinux
